gwtp-core/gwtp-mvp-client/src/main/java/com/gwtplatform/mvp/client/ViewImpl.java

```diff
@@ -32,17 +44,62 @@
  */
 public abstract class ViewImpl implements View {
     private Widget widget;
+    private Map<Object, HasOneWidget> oneWidgetSlots = new HashMap<Object, HasOneWidget>();
+
+    private Map<Object,HasWidgets> hasWidgetSlots = new HashMap<Object, HasWidgets>();
 
+    private Map<OrderedSlot<?>, List<Comparable<Comparable<?>>>> orderedSlots
+            = new HashMap<OrderedSlot<?>, List<Comparable<Comparable<?>>>>();
+
+    @SuppressWarnings("unchecked")
     @Override
     public void addToSlot(Object slot, IsWidget content) {
+        if (hasWidgetSlots.containsKey(slot)) {
+            if (orderedSlots.containsKey(slot)) {
```

Comment:
No idea why I did it this way.  Will fix.

On Thu, Feb 5, 2015 at 5:09 PM, Christopher Viel <notifications@github.com>
wrote:

> In
> gwtp-core/gwtp-mvp-client/src/main/java/com/gwtplatform/mvp/client/ViewImpl.java
> <https://github.com/ArcBees/GWTP/pull/672#discussion_r24168703>:
>
> >      @Override
> >      public void addToSlot(Object slot, IsWidget content) {
> > +        if (hasWidgetSlots.containsKey(slot)) {
> > +            if (orderedSlots.containsKey(slot)) {
>
> What do you think not to have the slots added to both hasWidgetSlots and
> orderedSlots? I think it would simplify a bit (ie: if(isInHasWidgetsSlots(slot)
> { /* do something */ } else if (isInOrderedSlots(slot)) { /* more stuff */ }
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ArcBees/GWTP/pull/672/files#r24168703>.
>